### PR TITLE
Fixes Bitbucket (cloud) Internal Server error on Jenkins

### PR DESCRIPTION
**Pull request by schowave**
Due to a invalid body length the payload cannot be delivered from the bitbucket cloud to the Jenkins server. This is fixed now, I could validate this behaviour using this fix.

### CODE REVIEW

**Comment by puneetverma05** [comment]
Hey, thanks for the PR @schowave, any update on when it will be getting merged?

**Comment by aphex3k** [comment]
@schowave Would you be able to resolve the conflict?

**Comment by xrecio** [comment]
> Hey, thanks for the PR @schowave, any update on when it will be getting merged?

in the meanwhile changes can be included in your /usr/lib/node_modules/smee-client/index.js and it will work.

**Review by ozkomer — APPROVED** [review]

**Comment by shadycraig** [comment]
Would be great to get this merged, anything I can do to help?

**Comment by theor** [comment]
yes please, can we help ?

**Comment by gr2m** [comment]
I'm reluctant about merging it because it's not tested. And I'm not sure if the content length is counted correctly. character length is not the same as byte length. 

If you could verify that it's working for different payloads, including all kind of unusual utf-8/utf-16 characters, that would help. If you could add tests that verify the implementation that would be even better

**Comment by mallox007** [comment]
Verified it does not work with special characters.  
Changed the call to 

`Buffer.byteLength(JSON.stringify(data.body))`

which resolves the problem

File: index.ts [hunk 1]
```diff
@@ -45,11 +45,14 @@ class Client {
 
     const req = superagent.post(url.format(target)).send(data.body)
 
+    const bodyLength = JSON.stringify(data.body).length
```
Comment (gr2m) [inline index.ts:48]:
```suggestion
    const bodyLength = Buffer.byteLength(JSON.stringify(data.body))
```

see https://github.com/probot/smee-client/pull/152#issuecomment-894154072

We should still add a test before merging this change

**Pull request closed without merging**